src/networking.c

```diff
@@ -3557,6 +3602,96 @@ void quitCommand(client *c) {
     c->flag.close_after_reply = 1;
 }
 
+int parseClientFilters(client *c, int i, clientFilter *filter) {
+    while (i < c->argc) {
+        int moreargs = c->argc > i + 1;
+
+        if (!strcasecmp(c->argv[i]->ptr, "id")) {
+            if (filter->ids == NULL) {
+                filter->ids = intsetNew(); // Initialize the intset for IDs
+            }
+            i++; // Move to the first ID after "ID"
+
+            // Process all IDs until a non-numeric argument or end of args
+            while (i < c->argc) {
+                long long id;
+                if (!string2ll(c->argv[i]->ptr, sdslen(c->argv[i]->ptr), &id)) {
+                    break; // Stop processing IDs if a non-numeric argument is encountered
```

Comment:
Ideally, it would work both ways with this implementation. If we want to restrict the IDs towards the end, then I can look into make some changes. 
Happy to let you and @madolson take the call on this.

Currently we have syntax error if non numeric, or unknown filter is passed.
```
127.0.0.1:6379> client list
id=3 addr=127.0.0.1:63139 laddr=127.0.0.1:6379 fd=8 name= age=48081 idle=46728 flags=N db=0 sub=0 psub=0 ssub=0 multi=-1 watch=0 qbuf=0 qbuf-free=0 argv-mem=0 multi-mem=0 rbs=1024 rbp=0 obl=0 oll=0 omem=0 tot-mem=1920 events=r cmd=client|list user=default redir=-1 resp=2 lib-name= lib-ver= tot-net-in=484 tot-net-out=215973 tot-cmds=13
id=8 addr=127.0.0.1:63183 laddr=127.0.0.1:6379 fd=9 name= age=48016 idle=48016 flags=N db=0 sub=0 psub=0 ssub=0 multi=-1 watch=0 qbuf=0 qbuf-free=0 argv-mem=0 multi-mem=0 rbs=1024 rbp=0 obl=0 oll=0 omem=0 tot-mem=1920 events=r cmd=command|docs user=default redir=-1 resp=2 lib-name= lib-ver= tot-net-in=27 tot-net-out=209291 tot-cmds=1
id=11 addr=127.0.0.1:64620 laddr=127.0.0.1:6379 fd=10 name= age=3 idle=0 flags=N db=0 sub=0 psub=0 ssub=0 multi=-1 watch=0 qbuf=0 qbuf-free=0 argv-mem=10 multi-mem=0 rbs=1024 rbp=0 obl=0 oll=0 omem=0 tot-mem=1946 events=r cmd=client|list user=default redir=-1 resp=2 lib-name= lib-ver= tot-net-in=53 tot-net-out=209291 tot-cmds=1
127.0.0.1:6379> client list id 3 str
ERR syntax error

```

Reply:
Yes, I know it's possible. The question is if there is some reason to forbid it, for example to make it less confusing for users or some tools. Personally, I think we can allow it.